src/messaging/ExchangeMgr.cpp

```diff
@@ -369,10 +369,13 @@ void ExchangeManager::OnConnectionExpired(SecureSessionHandle session, SecureSes
     }
 }
 
-void ExchangeManager::OnMessageReceived(const PacketHeader & header, const Transport::PeerAddress & source,
-                                        System::PacketBufferHandle msgBuf)
+void ExchangeManager::OnMessageReceived(const Transport::PeerAddress & source, System::PacketBufferHandle msgBuf)
 {
-    auto peer = header.GetSourceNodeId();
+    PacketHeader header;
+
+    ReturnOnFailure(header.DecodeAndConsume(msgBuf));
```

Comment:
Why decode it again ? isn't it already decoded in the `SecureSessionMgr` ? Can we pass it here ?